modules/calib3d/src/solvepnp.cpp

```diff
@@ -313,7 +314,13 @@ bool solvePnPRansac(InputArray _opoints, InputArray _ipoints,
     ipoints_inliers.resize(npoints1);
     try
     {
-        result = solvePnP(opoints_inliers, ipoints_inliers, cameraMatrix,
+       if (flags == SOLVEPNP_ITERATIVE && !useExtrinsicGuess)
```

Comment:
Do we need this condition ? Shouldn't that be for all methods ? Thx

Reply:
I think the issue only occurs for`SOLVEPNP_ITERATIVE` because AFAIK the other methods do not use extrinsic guess which was overwritten in each RANSAC iteration